lib/src/room.dart

```diff
@@ -581,4 +578,60 @@ class KurentoRoom extends ChangeNotifier implements Room {
       }
     }
   }
+
+  /// Creates and connects a track to send the [_localStream] to Kurento.
+  Future<void> _createOutgoingTrack() async {
+    Track track = await _client.createTrack(_serviceRequest.roomId, _serviceRequest.participantId);
+    _log.info('created outgoing track id=${track.id}');
+
+    _localTrackId = track.id;
+
+    await _connectTrack(_localTrackId!, _localStream);
+  }
+
+  /// Creates and connects a track to receive a remote stream from Kurento.
+  Future<void> _createIncomingTrack(int outgoingTrackId) async {
+    Track track = await _client.createTrack(_serviceRequest.roomId, _serviceRequest.participantId, outgoingTrackId);
+    _log.info('created incoming track id=${track.id} outgoing_track_id=$outgoingTrackId');
+
+    await _connectTrack(track.id);
+  }
+
+  Future<void> _reconnect() async {
+    try {
+      _log.info('reconnecting');
+
+      onReconnect?.call();
+
+      // Delete the old tracks.
+      await _client.deleteTracks(_serviceRequest.roomId, _serviceRequest.participantId);
+
+      // Close the old connections.
+      for (SfuConnection connection in _connectionByTrackId.values) {
+        await connection.dispose();
+      }
+      _connectionByTrackId.clear();
+
+      // Create and connect new tracks to receive the remote streams from the other participant(s). (Do this before our
+      // local stream, since we want to be able to hear the Agent ASAP.)
+      for (Participant participant in await _client.getParticipants(_serviceRequest.roomId)) {
+        if (participant.id != _serviceRequest.participantId) {
+          for (Track track in participant.tracks ?? []) {
+            if (track.isOutgoing) {
+              await _createIncomingTrack(track.id);
+            }
```

Comment:
I can see how this could make sense... it would be nice if all the functions were using the same point of reference. Here, the if condition `isOutgoing` seems to contradict the action `createIncomingTrack`.

Maybe the best solution to this issue is to use an "absolute" nomenclature (opposed to "relative" nomenclature) like `trackFromAgent` and `trackToAgent`.

Reply:
This is a side effect of how things are modeled in Platform. Tracks are either incoming or outgoing. An outgoing track is a send-only track from the participant to Kurento. An incoming track is a receive-only track from Kurento to the participant that maps to an outgoing track from another participant.